src/main/webapp/app/shared/markdown-editor/commands/courseArtifactReferenceCommands/channelMentionCommand.ts

```diff
@@ -0,0 +1,46 @@
+import { InteractiveSearchCommand } from 'app/shared/markdown-editor/commands/interactiveSearchCommand';
+import { faHashtag } from '@fortawesome/free-solid-svg-icons';
+import { HttpResponse } from '@angular/common/http';
+import { Observable, map, of } from 'rxjs';
+import { MetisService } from 'app/shared/metis/metis.service';
+import { ChannelIdAndNameDTO } from 'app/entities/metis/conversation/channel.model';
+import { ChannelService } from 'app/shared/metis/conversations/channel.service';
+
+export class ChannelMentionCommand extends InteractiveSearchCommand {
+    buttonIcon = faHashtag;
+
+    private cachedResponse: HttpResponse<ChannelIdAndNameDTO[]>;
+
+    constructor(
+        private readonly channelService: ChannelService,
+        private readonly metisService: MetisService,
+    ) {
+        super();
+    }
+
+    protected getAssociatedInputCharacter(): string {
```

Comment:
This could be a constant

Reply:
I prefer it to be a function since I can force the developer to implement it by declaring it abstract in the parent class.